mlflow/utils/file_utils.py

```diff
@@ -483,3 +485,54 @@ def _handle_readonly_on_windows(func, path, exc_info):
         raise exc_value
     os.chmod(path, stat.S_IWRITE)
     func(path)
+
+
+_TMP_DIR = None
+
+
+def get_or_create_tmp_dir():
+    """
+    Get or create a temporary directory which will be removed once python process exit.
+    """
+    from mlflow.utils.databricks_utils import is_in_databricks_runtime, get_repl_id
+
+    global _TMP_DIR
+
+    if _TMP_DIR is None:
+        if is_in_databricks_runtime() and get_repl_id() is not None:
+            # Note: For python process attached to databricks notebook, atexit does not work.
+            # The /tmp/repl_tmp_data/{repl_id} directory will be removed once databricks notebook detached.
+            _TMP_DIR = f"/tmp/repl_tmp_data/{get_repl_id()}"
+            os.makedirs(_TMP_DIR, exist_ok=True)
+        else:
+            _TMP_DIR = tempfile.mkdtemp()
+            atexit.register(shutil.rmtree, _TMP_DIR, ignore_errors=True)
+
+    return _TMP_DIR
+
+
+_TMP_NFS_DIR = None
+
+
+def get_or_create_nfs_tmp_dir():
+    """
+    Get or create a temporary NFS directory which will be removed once python process exit.
+    """
+    from mlflow.utils.databricks_utils import is_in_databricks_runtime, get_repl_id
+    from mlflow.utils.nfs_on_spark import get_nfs_cache_root_dir
+
+    global _TMP_NFS_DIR
+
+    nfs_root_dir = get_nfs_cache_root_dir()
+
+    if _TMP_NFS_DIR is None:
+        if is_in_databricks_runtime():
```

Comment:
Shall we also check that `get_repl_id() is not None` like we do in `get_or_create_tmp_dir()`?